app/views/projects/index.html.erb

```diff
@@ -1,39 +1,38 @@
-    <div id="page-content">
-    <h1 class="text-center mt-4">Listing all project</h1>
-    <div class="container">
-
-    
-
-    <% @projects.each do |project| %>
-        <div class="row justify-content-md-center">
-        <div class="col-8 mt-4">
-            <div class="card text-center shadow mb-5 bg-white rounded">
-            <div class="card-header font-italic">
-                    <h5 class="card-title"><%= link_to project.name, project_path(project), class: "text-success" %></h5>
-            </div>
-
-                    <div class="card-body">
-                    <%=  project.description %>
-                    <%= render_attachments(project.description) %>
-                    
+<div id="page-content">
+  <h1 class="text-center mt-4">Listing all projects</h1>
+  <div class="container">
+    <div id="projects">
+      <%= render @projects %>
+    </div>
+    <div class='dddd'>
+    </div>
+    <div id="pagination" class='hidden'>
+      <%= will_paginate @projects %>
+    </div>
+  </div>
+</div>
 
-                     <p class="card-text">Manger: <%= project.creator.name %></p>
-                    <p class="card-text">Members: <%= project.project_members.count %></p>
-                    <%= link_to "View", project_path(project), class: "btn btn-outline-success" %>
-                    <% if user_signed_in? %>
-                    <%= link_to "Edit", edit_project_path(project), class: "btn btn-outline-info" %>
 
-                    <%= button_to "Delete", project_path(project), method: :delete, data: {confirm: "Are you sure?"}, class: "btn btn-outline-danger" %>
+<script>
+  <% if @projects.next_page %>
+    var visitedUrl = undefined
+    $(window).on('scroll', function() {
+    // console.log($('.pagination a.next_page'));
+      var url = $('.pagination a.next_page').attr('href');
+      if (url && $(window).scrollTop() > $(document).height() - $(window).height() - 20) {
+        console.log('inside if scrolling');
+        console.log("url", url)
+        // $('.pagination').text('Loading more projects...');
+        if(visitedUrl !== url){
+            $.getScript(url);
+            visitedUrl = url
+        }
+      }
+    });
+  <% else %>
+    console.log('inside else');
+    $(window).off('scroll');
+    $('.pagination').remove();
+  <% end %>
+</script>
```

Comment:
@Usmanksri You do not need to add this in this file as you added the logic in the application.js you can remove it here. But from here you need to move the visted_url logic to application.js